metrics/metrics-api/src/main/java/org/apache/james/metrics/api/Metric.java

```diff
@@ -35,4 +35,15 @@
      * Implementation might be doing strict validation (throwing on negative value) or lenient (log and sanitize to 0)
      */
     long getCount();
+
+    /**
+     * Moving average.
+     *
+     * Period of the moving average is implementation defined.
```

Comment:
why? It should really depends on the domain

Reply:
I don't understand.

Reply:
BTW how to make naive metric implementation supporting moving averages (to avoid adding capabilities to the implementation) ?

Saying `period of the moving average is implementation specific` enables the `count over total time` naive implementation.

Reply:
> I don't understand.

Depending of what you measure you probably want to define the window of the average.
Moving average of a cache could be some minutes while moving average of a service call latency could be some seconds.
If you want to make it the simplest possible, you should at least define what is the fixed value used for the current implementation.
Anybody looking at the doc could then decide whether it fits her need or not.